src/inertia.js

```diff
@@ -24,6 +26,15 @@ export default {
       this.setPage(initialPage)
     }
 
+    this.saveScrollPositions = Debounce(() => {
+      this.setState({
+        ...window.history.state,
+        scrollRegions: Array.prototype.slice.call(
+          document.querySelectorAll('[scroll-region]')
+        ).map(region => region.scrollTop),
```

Comment:
While we're at it, why not save `scrollLeft` as well?

Reply:
Done. 👍